### PR TITLE
i18n support

### PR DESCRIPTION
**Pull request by grezniczek**
@mmcev106 
As per your suggestion, I created a new branch (based upon the most recent `testing` https://github.com/vanderbilt/redcap-external-modules/commit/2ae995b66cd0be33056842f3e4100be236c80565 from `vanderbilt/redcap-external-modules`) and copied over the files from the latest version of the `Language feature` pull request.

I went through all the files on a line by line basis and only retained the changes necessary to support the new features and the translation of the EM framework (I did not split i18n support and framework translation into two separate pull requests after all - that would have been much work). 

Where necessary, I updated the translations (or added new ones) to be up-to-date with changes introduced into `testing` in the meantime. 

There may be 1 or 2 very minor bug fixes that I retained as well, such as the method signature of _ExternalModules::getDatacoreEmails()_ (my tooling was complaining as it is called without arguments).

I did commit the changes in ~20 commits, but you could squash those for the merge if you want.

I probably should have started with a brand new fork ... but I haven't and thus there are some old commits from before today in the list (I accidentally commited my first changes to `testing`, but reverted them after creating a separate branch - so those should be "neutral").

I did some testing in REDCap 9.3.8 and all seems to work as expected. Unfortunately, on the machine I am on now, I cannot run the tests.

_Just as a reminder:_  
While the strings used by the EM framework have been extracted and put into `classes/English.ini`, there is no mechanism in place that would allow these to be translated. The intended mechanism still is to have these merged with **REDCap**'s `English.ini` upon release packaging (and removing `classes/English.ini` in the process).

### CODE REVIEW

**Comment by grezniczek** [comment]
Before you merge ....
I found that one of my modules is causing an error to be generated only in the modified framework:

> Could not instantiate the 'redcap_instancetype_indicator' module, but a specific cause could not be detected. This could be caused by a die() or exit() call in the module which needs to be replaced with an exception to provide more details, or a $module->exitAfterHook() call to allow other modules to execute for the current hook.

This is only happening on the External Module page in the project context (not system). I have already stripped down the module to doing nothing. I guess it must have to do with the configuaration, but there is nothing special there ... the module is on [GitHub](https://github.com/grezniczek/redcap_instancetype_indicator). I will keep investigating ....

File: manager/templates/enabled-modules.php [hunk 1]
```diff
@@ -218,6 +229,8 @@
 		echo 'None';
 	} else {
 		foreach ($versionsByPrefix as $prefix => $version) {
+			// Ensure module has been instantiated so that translations are available.
+			ExternalModules::getModuleInstance($prefix);
```
[the remark this answers is not in the document]
Comment (grezniczek) [reply manager/templates/enabled-modules.php]: Actually I just came to this as the cause for the error mentioned above.

Your suggestion is excellent - **and** fixes the error (though I still have no idea, why only this particular module showed up in the error email).

https://github.com/vanderbilt/redcap-external-modules/pull/219/commits/9e8f080fac8e977666b5fb29a1642b21ece05b34

**Comment by mmcev106** [comment]
This might be ready to merge!  I'm going to do a little more testing and have one other person review it on my end.  Will report back soon.

**Comment by mmcev106** [comment]
@grezniczek, this has been merged!  Should we create an issue to track @kcmcg's question?

This will likely move to our production server on the 18th, and will likely be included in the next REDCap release after the 25th.  Thanks you so much for the hard work!  I'm hopeful this will bring a lot of value to many people.

**Pull request merged**